cmd/daprd/main.go

```diff
@@ -97,6 +97,15 @@ func main() {
 		log.Fatal(err)
 	}
 
+	stop := make(chan os.Signal, 1)
+	signal.Notify(stop, syscall.SIGTERM, os.Interrupt)
+
+	go func() {
+		// handle signals and perform any cleanup before exiting
+		cleanup(rt, stop)
```

Comment:
We should call `cleanup` once, calling the method `cleanup` in line `280` is misleading, as it's not a cleanup method in that context, its blocking and then might clean up..

Reply:
Look at how [Sentry](https://github.com/dapr/dapr/blob/c5ac112253968bb37bb89e2156e27487c98b73b7/pkg/sentry/sentry.go#L47) is dealing with it, we should probably do something similar.